app/bibleview-js/src/components/TextEditor.vue

```diff
@@ -103,18 +103,50 @@ export default {
       title: 'Insert bible reference',
       result: async () => {
         const originalRange = document.getSelection().getRangeAt(0);
+        //Get the active languages and create a bible reference parser for each language
+        const languages = android.getActiveLanguages()
+        //Always add the "en" language, so at least the english parser always exists.
+        if (!languages.includes("en")) {
+            languages.push("en")
+        }
+        var parsers = []
+        languages.forEach(async (lang) => {
+            try {
+                const bcv_parser = await import(`bible-passage-reference-parser/js/${lang}_bcv_parser`)
+                    .then((module) => {
+                        return new module.bcv_parser
+                    });
+                parsers.push(bcv_parser)
+            } catch (error) {
+                console.log(`No parser exists for language: ${lang}`)
+            }
```

Comment:
also this importing I think we should do only once and use cached parsers from then on.

Reply:
If module language is available (for all new bookmarks it is), I think we could try to use module language and UI language instead of trying everything?

Reply:
I tried to get the module language, but I couldn't find out how. Maybe you can shed some light on this.

Reply:
If using bookmark: add language info to ClientBookmark and pass it to TextEditor component via props or via provide()/inject(). 
If using document (for bibleview), add info to OsisFragment. (other options OsisDocument / BibleDocument) but I believe OsisFragment is good place; then it can be read on client side by injecting osisFragment).

Reply:
The module language should only take priority over the others, not replace it. Since I might want to make a Dutch comment on the KJV bible, and include a Dutch reference.

Reply:
I'll skip this for now, using all parsers does not have any performance impact afaict.
In addition, getting the module language is tricky and it would require that we load parsers dynamically instead of only once.